File: backend/src/domain/transfer-status.ts
/** Lifecycle of a transfer. Closed set as `as const` — compared/serialized by name. */
export const TransferStatus = {
  PENDING: "PENDING",
  COMPLETED: "COMPLETED",
  FAILED: "FAILED",
  EXPIRED: "EXPIRED",
} as const;

export type TransferStatus = (typeof TransferStatus)[keyof typeof TransferStatus];

File: app/lib/features/transfer/domain/entities/transfer_status.dart
enum TransferStatus {
  pending,
  completed,
  failed,
  expired;

  static TransferStatus fromWire(String value) {
    return switch (value) {
      'PENDING' => TransferStatus.pending,
      'COMPLETED' => TransferStatus.completed,
      'FAILED' => TransferStatus.failed,
      'EXPIRED' => TransferStatus.expired,
      _ => throw ArgumentError.value(value, 'value', 'Unknown status'),
    };
  }
}
